src/client/push-client.js

```diff
@@ -16,24 +16,24 @@ import SubscriptionFailedError from './subscription-failed-error';
 import PushClientEvent from './push-client-event';
 import EventDispatch from './event-dispatch';
 
-// document.currentScript is not supported in all browsers, but it IS supported
-// in all browsers that support Push.
-// TODO(mscales): Ensure that this script will not cause errors in unsupported
-// browsers.
-let currentScript = document.currentScript.src;
+// Set the default scope to be relative to the service worker
+// script with an addition string to prevent any unlikely collisions
+// This should allow multiple instances of push on the same scope to work
+const SCOPE = './goog.push.scope/';
 
-// Make the dummy service worker scope be relative to the library script. This
-// means that you can have multiple projects hosted on the same origin without
-// them interfering with each other, as long as they each use a different URL
-// for the script.
-const SCOPE = new URL('./goog.push.scope/', currentScript).href;
-const WORKER_URL = new URL('./worker.js', currentScript).href;
 const SUPPORTED = 'serviceWorker' in navigator &&
     'PushManager' in window &&
     'Notification' in window &&
     'permissions' in navigator &&
     'showNotification' in ServiceWorkerRegistration.prototype;
 
+const ERROR_MESSAGES = {
+  'bad constructor': 'The PushClient constructor expects either service ' +
+    'worker registration or the path to a service worker file and an ' +
+    'optional scope string.',
+  'redundant worker': 'Worker became redundant'
+};
+
 const registrationReady = function(registration) {
   if (registration.active) {
```

Comment:
This library only cares that the user can be subscribed / unsubscribed. Which service worker shouldn't matter but the subscription will transfer (or least it should do).

Is there any scenario where the latest service is required?